### PR TITLE
Fix issue 38095 TODOs in gloo tests

### PR DESCRIPTION
**Pull request by kit1980**
Fix TODOs related to https://github.com/pytorch/pytorch/issues/38095

### CODE REVIEW

File: test/distributed/test_c10d_gloo.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def simple_coalesced_reduce_tests(rank, world_size):
         (
             c10d.ReduceOp.MAX,
             [torch.tensor([rank + x]) for x in [1.0, 2.0]],
-            [torch.tensor([world_size]), torch.tensor([world_size + 1.0])],
+            [torch.tensor([world_size + 0.0]), torch.tensor([world_size + 1.0])],
```
Comment (ZainRizvi) [inline test/distributed/test_c10d_gloo.py:155]: nit: any difference between `world_size + 0.0` here and the `float(world_size)` on line 78? 

Wondering if one should be changed to the other for consistency

File: test/distributed/test_c10d_gloo.py [hunk 2]
```diff
@@ -526,12 +519,11 @@ def _test_allreduce_stress(self, inputs):
...
         for i, future_handle in enumerate(future_handles):
             future_handle.wait()
-            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-            self.assertEqualIgnoreType(
+            self.assertEqual(
                 torch.tensor(
                     [
                         (i * self.world_size)
-                        + (self.world_size * (self.world_size - 1) / 2)
+                        + (self.world_size * (self.world_size - 1) // 2)
```
Comment (ZainRizvi) [inline test/distributed/test_c10d_gloo.py:526]: cool, didn't know this was a thing!

**Review by ZainRizvi — APPROVED** [review]

File: test/distributed/test_c10d_gloo.py (hunk 1 above)
Comment (kit1980) [reply test/distributed/test_c10d_gloo.py]: Updated.

**Pull request closed without merging**